linode/manager
# refactor: [M3-7377] - Migrate dopdowns in Create Firewall Drawer from Autocomplete to LinodeSelect and NodeBalancerSelect

## Description 📝
Migrated dopdowns in Create Firewall Drawer from Autocomplete to LinodeSelect and NodeBalancerSelect.

## Changes  🔄
List any change relevant to the reviewer.
- Migrated dopdowns in Create Firewall Drawer from Autocomplete to LinodeSelect and NodeBalancerSelect.
- Updated LinodeSelect to use new autocomplete component: `src/components/Autocomplete/Autocomplete`
- Updated NodeSelect to use new autocomplete component
- Added unit tests (based on `LinodeSelect.test.tsx`) for `NodeBalancerSelect.tsx`

## Preview 📷
**Include a screenshot or screen recording of the change**

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-11-09 at 11 23 55 AM](https://github.com/linode/manager/assets/139489745/7905bdd4-0f88-483c-8a7b-c71f0a810b8b) | ![Screenshot 2023-11-15 at 2 27 10 PM](https://github.com/linode/manager/assets/139489745/c79202fa-8ea4-4e21-8770-1074ec0b1155) |
| ![Screenshot 2023-11-09 at 11 24 22 AM](https://github.com/linode/manager/assets/139489745/36ed9559-84bc-4c24-a61f-48d8f66bff44) | ![Screenshot 2023-11-15 at 2 27 56 PM](https://github.com/linode/manager/assets/139489745/6912b4ec-0dec-4b9f-b1a1-6431c3e286e0) |

## How to test 🧪

### Prerequisites
(How to setup test environment)
- There are no prerequisites

### Reproduction steps
(How to reproduce the issue, if applicable)
- Since this is a refactor, there are no reproduction steps

### Verification steps 
(How to verify changes)

- [ ] Navigate to http://localhost:3000/linodes/create and ensure when clicking on 'Create Firewall' in the Firewall Panel, the label for the Linodes dropdown is 'Additional Linodes (Optional)'
- [ ] Navigate to http://localhost:3000/nodebalancers/create and ensure when clicking on 'Create Firewall' in the Firewall Panel, the label for the NodeBalancers dropdown is 'Additional NodeBalancers (Optional)'

Ensure the LinodeSelect and NodeBalancerSelect dropdowns are working:
- [ ] http://localhost:3000/firewalls/create
- [ ] http://localhost:3000/domains/create

## As an Author I have considered 🤔

*Check all that apply*
